code/game/atoms.dm

```diff
@@ -1057,3 +1077,18 @@
 
 /atom/proc/intercept_zImpact(atom/movable/AM, levels = 1)
 	return FALSE
+
+///Sets the custom materials for an item.
+/atom/proc/set_custom_materials(var/list/materials)

```

Comment:
I feel like this proc should live with the custom materials or the subsystem

Reply:
This is for mass replacing all the materials in a thing and seems more associated with the thing getting affected than any particular material.